src/main/memory.c

```diff
@@ -646,7 +646,15 @@ static R_size_t R_NodesInUse = 0;
   case WEAKREFSXP: \
   case RAWSXP: \
   case S4SXP: \
+    break; \
   case EXTERNALSXP: \
+    { \
+      int i; \
+      SEXP* end = (SEXP*)((uintptr_t)INTEGER(__n__) + LENGTH(__n__)); \
+      SEXP* ptrs = (SEXP*)((uintptr_t)end - (TRUELENGTH(__n__)*sizeof(SEXP))); \
```

Comment:
To my eyes it would be more ideomatic to say: 
```
SEXP* ptrs = end - TRUELENGTH(__n__);
```

Reply:
actually, thinking about it, your version is even wrong. it should be `sizeof(SEXP*)`. Of course on probably every platform out there `sizeof(SEXP*) == sizeof(SEXP)` :)

Reply:
Whoops, thanks for the catch!